### PR TITLE
Panic Bunker

### PR DESCRIPTION
**Pull request by AffectedArc07**
:wave:

**What does this PR do:**
This PR adds the panic bunker as an admin toggle. Why? Because we have a 200k tubetide coming in about 5 hours.

**Changelog:**
:cl: AffectedArc07
add: Panic Bunker
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
So what does it...actually do?

**Comment by AffectedArc07** [comment]
@TDSSS Stops people who havnt played on the server before connecting. This stops the server buckling to a crawl and stops the admins from going insane when a tubetide hits.

**Comment by Spartan6** [comment]
Administrators: The perfect PR does not exist.

AffectedArc07:

**Comment by AffectedArc07** [comment]
Welp this wasnt merged in time

F

**Comment by taukausanake** [comment]
Isn't it called going unlisted? Pretty sure people aren't that smart to go to the Paradise homepage

**Comment by AffectedArc07** [comment]
@taukausanake Going wont solve the issue. Paradise is a main target for tube-tides due to being the largest server. People go to the website and see Paradise as the largest one, then they use the linked button on the SS13 website to join. Panic bunker allows us to stay listed but stop the server and admins from dying.

**Comment by Kyep** [comment]
Briefly:
- We have open feature PRs from ~60 days ago. Expecting a newly created feature PR to be merged within 5 hours is not realistic. This is double true when we usually require at least a 24 hour waiting period to merge feature PRs. While we appreciate the prompt response, expecting it to be merged within 5 hours is not a realistic expectation.
- Rimmy has done 2 other videos that mentioned SS13 (radio 1w ago, barotrauma 2w ago). Those videos had ~100k views, which is not in the same league as sseth (>1.5M views). Their communities are also very different - we haven't observed any substantial issues with Rimmytide thus far.
- While a panic bunker is not necessarily a bad idea, personally I would lean more towards an automated system (e.g: automatic panic bunker at ~150 concurrent connections) rather than something that has to be manually activated/deactivated. That way if say a popular streamer starts streaming paradise the system can react almost instantly.

**Comment by AffectedArc07** [comment]
Panic bunker now auto-engages after 150 players are hit, but will not auto disable in case admins set it for a specific reason. Unfortunately I wasn't able to test it auto enabling at 150 players, but the code says it should work

*famous last words*

**Comment by Kyep** [comment]
The number of players it auto engages at (if it does) should be a config setting, in case we need to change it in future. The auto-engaging past a threshold is helpful, though.

File: code/modules/client/client procs.dm [hunk 1]
```diff
@@ -537,6 +547,14 @@
 			message_admins("SQL ERROR during log_client_to_db (update). Error : \[[err]\]\n")
 	else
 		//New player!! Need to insert all the stuff
+
+		// Check new peeps for panic bunker
+		if(GLOB.panic_bunker_enabled)
+			message_admins("<span class='adminnotice'>Failed Login: [key] - New account attempting to connect during panic bunker</span>")
```
Comment (Ty-Omaha) [inline code/modules/client/client procs.dm]: Seems like it will cause a lot of log spam

File: code/modules/client/client procs.dm [hunk 2]
```diff
@@ -537,6 +547,14 @@
...
+			src << "Sorry but the server is currently not accepting connections from never before seen players."
```
Comment (Ty-Omaha) [inline code/modules/client/client procs.dm]: Adding a "Try again later" would be good, with the automatic stuff kyet said

Comment (AffectedArc07) [reply code/modules/client/client procs.dm]: The automatic stuff is in, ill change this message

File: code/modules/client/client procs.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/modules/client/client procs.dm]: So do many other things, such as attack logs. This isnt that big of an issue

Comment (Ty-Omaha) [reply code/modules/client/client procs.dm]: Attack logs have levels where we can filter out what is usually just spam. 

Why do admins need to be notified of each individual person? We're already notified that the panic bunker is on.

Comment (AffectedArc07) [reply code/modules/client/client procs.dm]: So they can prepare for specific people questioning why they cant get in? This is how TG did it so.........

**Pull request merged**